### PR TITLE
Make tile reusable

### PR DESCRIPTION
**Pull request by cquiroz**
As discussed in a previous PR we could enhance tiles to support `Reusability` via `Reusable`. I maybe even able to understand what's going on

### CODE REVIEW

File: common/src/main/scala/explore/components/Tile.scala [hunk 1]
```diff
@@ -41,8 +39,7 @@ object Tile {
 
   type RenderInTitle = VdomNode ~=> VdomNode
 
-  // Explicitly never reuse as we are not considering the content
-  implicit val propsReuse: Reusability[Tile] = Reusability.never
+  implicit val propsReuse: Reusability[Tile] = Reusability.derive
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:42]: This is probably not taking the `render` parameter into account. I think we can add ` && Reusability.by(_.render)` at the end.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 2]
```diff
@@ -212,6 +210,25 @@ object ObsTabContents {
         }.getOrEmpty
       }
 
+    def constraintsSelectorFn(
+      constraintsSetId: Option[ConstraintSet.Id],
+      obsSummaryOpt:    Option[ObsSummary],
+      observations:     ConstraintsInfo
+    )(implicit
+      client:           clue.TransactionalClient[IO, explore.schemas.ObservationDB]
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:218]: This is great. I hadn't realized that converting a method to a function with `fn _` resolves the implicits. I was jumping through hoops all over to pass implicits to these render functions. I can simplify all that now.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: common/src/main/scala/explore/components/Tile.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/components/Tile.scala:42]: I wonder if this is still the case in Scala 3, which has context functions. I'll give it a try before changing anything.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala (hunk 2 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/tabs/ObsTabContents.scala:218]: I wonder if this is still the case in Scala 3, which has context functions. I'll check that before changing anything.

**Review by toddburnside — APPROVED** [review]
👍

File: common/src/main/scala/explore/components/Tile.scala (hunk 1 above)
Comment (cquiroz) [reply common/src/main/scala/explore/components/Tile.scala:42]: Shouldn't `derive` take into account all fields?

Comment (rpiaggio) [reply common/src/main/scala/explore/components/Tile.scala]: I'm not sure how the macro is implemented. There is a possibility it only takes a look at the first parameter group.

**Pull request merged**